eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -155,27 +163,31 @@ jobs:
         vcpkg.deps: 'cmocka'
         VCPKG_DEFAULT_TRIPLET: 'x64-linux'
         build.args: ' -DUNIT_TESTING=ON -DPRECONDITIONS=OFF'
+        AZ_SDK_C_NO_SAMPLES: 'true'
       Win_x86_with_unit_test_no_preconditions:
         vm.image: 'windows-2019'
         vcpkg.deps: 'cmocka'
         VCPKG_DEFAULT_TRIPLET: 'x86-windows-static'
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: Win32
         build.args: ' -DUNIT_TESTING=ON -DPRECONDITIONS=OFF'
+        AZ_SDK_C_NO_SAMPLES: 'true'
       Win_x64_with_unit_test_no_preconditions:
         vm.image: 'windows-2019'
         vcpkg.deps: 'cmocka'
         VCPKG_DEFAULT_TRIPLET: 'x64-windows-static'
         CMAKE_GENERATOR: 'Visual Studio 16 2019'
         CMAKE_GENERATOR_PLATFORM: x64
         build.args: ' -DUNIT_TESTING=ON -DPRECONDITIONS=OFF'
+        AZ_SDK_C_NO_SAMPLES: 'true'
```

Comment:
Why aren't we building the samples most of the time?

Reply:
We have an specific entry in the matrix to build samples:
https://github.com/Azure/azure-sdk-for-c/blob/master/eng/pipelines/templates/jobs/archetype-sdk-client.yml#L49

It was a bug that we were building samples in all CI pipeline configurations. That's why I added this missing env var to all configs were we are not expecting to build samples

Reply:
Got it.